Frontend/src/components/calendar/CalendarModal.jsx

```diff
@@ -0,0 +1,138 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { UserContext } from '../../UserContext.jsx';
+import { useNavigate } from 'react-router-dom';
+import '../mentor/mentor-profile/MentorProfileModal.css'
+import config from '../../../config.js';
+
+function CalendarModal ({toggleModal, onMeetingScheduled}) {
+    const {user} = useContext(UserContext);
+    const [menteesOrMentors, setMenteesOrMentors] = useState([]);
+    const [selectedUser, setSelectedUser] = useState('');
+    const [scheduledDate, setScheduledDate] = useState('');
+    const [scheduledTime, setScheduledTime] = useState('');
+    const [topic, setTopic] = useState('');
+    const [errorMessage, setErrorMessage] = useState('');
+    const [userData, setUserData] = useState('');
+    const navigate = useNavigate();
+
+    // Fetch user data (mentor or mentee)
+    useEffect(() => {
+        const fetchUserData = async () => {
+            try {
+                const url = user.userRole === 'mentor'
+                    ? `${config.apiBaseUrl}/mentors/${user.id}`
+                    : `${config.apiBaseUrl}/mentees/${user.id}`;
+
+                const response = await fetch(url);
+                if (!response.ok) {
+                    throw new Error('Failed to fetch user data');
+                }
+                const data = await response.json();
+                setUserData(user.userRole === 'mentor' ? data.mentor : data.mentee);
+            } catch (error) {
+                setErrorMessage(error.message);
+            }
+        };
+
+        if (user && user.id) {
+            fetchUserData();
+        }
+    }, [user]);
+
+    // Fetch list of requests using mentor ID
+    useEffect(() => {
+        const fetchRequests = async (userId) => {
+        try {
+            const url = user.userRole === 'mentor'
+                ? `${config.apiBaseUrl}/connect-requests/${userId}`
+                : `${config.apiBaseUrl}/connect-requests/mentee/${userId}`
+
+            const response = await fetch(url);
+            if (!response.ok) {
+                throw new Error('Failed to fetch requests');
+            }
+            const data = await response.json();
+            const acceptedRequests = data.requests.filter(request => request.status === 'accepted');
+            setMenteesOrMentors(acceptedRequests.map(request => ({
+                id: user.userRole === 'mentor' ? request.menteeId : request.mentorId,
+                name: user.userRole === 'mentor' ? request.menteeName : request.mentorName
+            })));
```

Comment:
Can a single user be both a mentor and a mentee? or would that necessarily require 2 accounts?

Reply:
A single user can be only a mentor or a mentee, so it requires 2 separate accounts.